src/map/mob.cpp

```diff
@@ -4194,248 +4197,938 @@ static void item_dropratio_adjust(t_itemid nameid, int mob_id, int *rate_adjust)
 	}
 }
 
-/*==========================================
- * processes one mobdb entry
- *------------------------------------------*/
-static bool mob_parse_dbrow(char** str)
-{
-	struct mob_db *db, entry;
-	struct status_data *status;
-	int mob_id, i;
-	double exp, maxhp;
-	struct mob_data data;
+const std::string MobDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/mob_db.yml";
+}
 
-	mob_id = atoi(str[0]);
+bool MobDatabase::parseDropNode(std::string nodeName, YAML::Node node, uint8 max, s_mob_drop *drops) {
+	const YAML::Node &dropNode = node[nodeName];
+	uint16 i = 0;
+
+	for (const YAML::Node &dropit : dropNode) {
+		if (i >= max) {
```

Comment:
Still wrong :(
Move index calculation up here and only check in the case where you calculate i